Libraries/LibWeb/DOM/Slottable.cpp

```diff
@@ -178,12 +178,6 @@ void assign_slottables(GC::Ref<HTML::HTMLSlotElement> slot)
 // https://dom.spec.whatwg.org/#assign-slotables-for-a-tree
 void assign_slottables_for_a_tree(GC::Ref<Node> root)
 {
-    // AD-HOC: This method iterates over the root's entire subtree. That iteration does nothing if the root is not a
-    //         shadow root (see `find_slottables`). This iteration can be very expensive as the HTML parser inserts
-    //         nodes, especially on sites with many elements. So we skip it if we know it's going to be a no-op anyways.
-    if (!root->is_shadow_root())
```

Comment:
I'm trying to figure out if it's _only_ the "relevant agent’s signal slots" event firing that cares about this, in which case, I wonder if we can come up to detect a different metric which can figure out if we need to fire any `slotchange` events so we can keep this fast path 🤔